miso-web/src/main/webapp/scripts/library_hot.js

```diff
@@ -8,12 +8,16 @@ Library.hot = {
   libraryTypeAliases: {},
   
   getLibraryTypeAliasLists: function () {
-    Hot.dropdownRef.libraryTypes.forEach(function (lt) {
-      if (Library.hot.libraryTypeAliases[lt.platform]) {
-        Library.hot.libraryTypeAliases[lt.platform].push(lt.alias);
-      } else {
-        Library.hot.libraryTypeAliases[lt.platform] = [lt.alias];
+    var usedTypes = Library.hot.librariesJSON.map(function (lib) {
+      return lib.libraryTypeId;
+    });
+    Hot.dropdownRef.libraryTypes.filter(function (lt) {
+      return !lt.archived || usedType.indexOf(lt.id) != -1;
+    }).forEach(function (lt) {
+      if (!Library.hot.libraryTypeAliases[lt.platform]) {
+        Library.hot.libraryTypeAliases[lt.platform] = [];
       }
+      Library.hot.libraryTypeAliases[lt.platform].push(lt.alias);
```

Comment:
Bulk create libraries is fine, but for bulk edit libraries, after `Library.hot.getLibraryTypeAliasLists()` is called on `bulkEditLibraries.jsp`, you'll need to go through each LibraryType cell and update its source to `Library.hot.libraryTypeAliases[${platformName}]`. In bulk edit, no source is being set for library type.